packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -325,4 +326,30 @@ export default class ContactCenter extends WebexPlugin implements IContactCenter
       throw detailedError;
     }
   }
+
+  /**
+   * Handles the device type specific logic
+   */
+  private async handleDeviceType(deviceType: string, dn: string): Promise<void> {
+    switch (deviceType) {
+      case LoginOption.BROWSER:
+        await this.webCallingService.registerWebCallingLine();
+        this.agentConfig.deviceType = LoginOption.BROWSER;
+        break;
+      case LoginOption.AGENT_DN:
+        this.agentConfig.deviceType = LoginOption.AGENT_DN;
+        this.agentConfig.defaultDn = dn;
+        break;
+      case LoginOption.EXTENSION:
+        this.agentConfig.deviceType = LoginOption.EXTENSION;
+        this.agentConfig.defaultDn = dn;
+        break;
+      default:
+        LoggerProxy.error(`Unsupported device type: ${deviceType}`, {
```

Comment:
It should ideally never reach here, however we must account for all possibilites.